### PR TITLE
api-server: get robots

### PR DESCRIPTION
**Pull request by koonpeng**
## What's new

Adds new endpoints/more useful data in existing fleets endpoint

* `/fleets` now supports querying, the returned data now includes the current state.
* new `/fleets/robots` endpoint, the endpoint supports querying, the return data includes robot's fleet, robot state, and currently active, pending and queued tasks assigned to the robot.

## Self-checks

- [ ] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

File: packages/api-server/api_server/routes/fleets.py [hunk 1]
```diff
@@ -1,22 +1,97 @@
...
+                    fleet=q.fleet_name, name=q.robot_name, state=q.data
+                )
+                for q in robot_states.items
+            }
+
+            filter_states = [
+                TaskStateEnum.ACTIVE.value,
+                TaskStateEnum.PENDING.value,
+                TaskStateEnum.QUEUED.value,
+            ]
```
Comment (koonpeng) [inline packages/api-server/api_server/routes/fleets.py:76]: @vallq Is this the correct task states that should be displayed on the robot panels?

File: packages/api-server/api_server/models/doors.py [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
+from tortoise.contrib.pydantic.creator import pydantic_model_creator
 
 from . import tortoise_models as ttm
 from .ros_pydantic import rmf_building_map_msgs, rmf_door_msgs
 
 Door = rmf_building_map_msgs.Door
 DoorMode = rmf_door_msgs.DoorMode
 DoorState = rmf_door_msgs.DoorState
-DoorHealth = ttm.DoorHealth
+DoorHealth = pydantic_model_creator(ttm.DoorHealth)
```
Comment (matiasbavera) [inline packages/api-server/api_server/models/doors.py:10]: Just to check, can this use `from .health import BasicHealth` like ingestors and fleet?

File: packages/api-server/api_server/models/lifts.py [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 
 from pydantic import BaseModel, Field
+from tortoise.contrib.pydantic.creator import pydantic_model_creator
 
 from . import tortoise_models as ttm
 from .ros_pydantic import rmf_building_map_msgs, rmf_lift_msgs
 
 Lift = rmf_building_map_msgs.Lift
-LiftHealth = ttm.LiftHealth
+LiftHealth = pydantic_model_creator(ttm.LiftHealth)
```
Comment (matiasbavera) [inline packages/api-server/api_server/models/lifts.py:10]: Just to check, can this use from .health import BasicHealth like ingestor and fleet?

**Review by matiasbavera — COMMENTED** [review]
Thanks for this, left some comments related to the health type

File: packages/api-server/api_server/models/doors.py (hunk 2 above)
Comment (koonpeng) [reply packages/api-server/api_server/models/doors.py:10]: this change is because the type checker/lint have problems with the old code, it doesn't infer the types correctly when using language features like class decorations.

**Review by matiasbavera — APPROVED** [review]

File: packages/api-server/api_server/routes/fleets.py (hunk 1 above)
Comment (vallq) [reply packages/api-server/api_server/routes/fleets.py:76]: hi KP, sorry for the delayed reply!! yes, just these 3 tasks states
